No explicit implementation for ‘generalBracket’

`exceptions-0.10` added a `generalBracket` member to `MonadMask`.
(`exceptions-0.9` is not worth supporting - it has different and broken `generalBracket`).

```haskell
src/Control/Monad/Trans/AWS.hs:212:10: warning: [-Wmissing-methods]
    • No explicit implementation for
        ‘generalBracket’
    • In the instance declaration for ‘MonadMask (AWST' r m)’
    |
212 | instance MonadMask m => MonadMask (AWST' r m) where
```

https://github.com/brendanhay/amazonka/blob/248f7b2a7248222cc21cef6194cd1872ba99ac5d/amazonka/src/Control/Monad/Trans/AWS.hs#L212

Should be easy to fix.

--- a/amazonka/src/Control/Monad/Trans/AWS.hs
+++ b/amazonka/src/Control/Monad/Trans/AWS.hs
@@ -1,5 +1,6 @@
 {-# LANGUAGE BangPatterns               #-}
 {-# LANGUAGE ConstraintKinds            #-}
+{-# LANGUAGE CPP                        #-}
 {-# LANGUAGE FlexibleContexts           #-}
 {-# LANGUAGE FlexibleInstances          #-}
 {-# LANGUAGE GeneralizedNewtypeDeriving #-}
@@ -215,6 +216,15 @@ instance MonadMask m => MonadMask (AWST' r m) where
 
     uninterruptibleMask a = AWST' $ uninterruptibleMask $ \u ->
         unAWST $ a (AWST' . u . unAWST)
+
+#if MIN_VERSION_exceptions(0,10,0)
+    generalBracket acquire rel action = AWST' $
+        generalBracket
+            (unAWST acquire)
+            (\a ex -> unAWST $ rel a ex)
+            (\a -> unAWST $ action a)
+#endif
+
 
 instance MonadBase b m => MonadBase b (AWST' r m) where
     liftBase = liftBaseDefault

